src/core.ts

```diff
@@ -151,7 +151,7 @@ function getExposedContent(ast: SourceUnit, relativizePath: (p: string) => strin
         const subset: Visibility[] = isLibrary ? [ 'internal', 'public', 'external' ] : ['internal'];
 
         const hasReceiveFunction = getFunctions(c, deref, [ 'external' ]).some(fn => fn.kind === 'receive');
-        const externalizableVariables = getVariables(c, deref, subset).filter(v => v.typeName?.nodeType !== 'UserDefinedTypeName' || isTypeExternalizable(v.typeName, deref));
+        const externalizableVariables = getVariables(c, deref, subset).filter(v => (v.typeName?.nodeType !== 'UserDefinedTypeName' && v.typeName?.nodeType !== 'ArrayTypeName') || isTypeExternalizable(v.typeName, deref));
```

Comment:
It seems like this should be equivalent... do you see a reason why we wouldn't do this?

```suggestion
        const externalizableVariables = getVariables(c, deref, subset).filter(v => isTypeExternalizable(v.typeName, deref));
```

Reply:
Yeah this doesn't seem to work.